app/src/main/AndroidManifest.xml

```diff
@@ -27,41 +27,41 @@
     <uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT" />
     <uses-permission android:name="com.android.alarm.permission.SET_ALARM" />
 
-    <uses-permission-sdk-23 android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" />
-
-    <!-- To receive data from xdrip. -->
+    <uses-permission-sdk-23 android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" /> <!-- To receive data from xdrip. -->
```

Comment:
Too much formatting changes in files. Try to change diff to contain only changes needed by new code